modules/datadict/jsx/dataDictIndex.js

```diff
@@ -0,0 +1,112 @@
+import FilterForm from 'FilterForm';
+import formatDataDictColumn from './columnFormatter';
+
+/**
+ * Data Dictionary Page.
+ *
+ * Serves as an entry-point to the module, rendering the whole react
+ * component page on load.
+ *
+ * Renders Data Dictionary main page consisting of FilterTable and
+ * DataTable components.
+ *
+ * @author Liza Levitis
+ * @version 1.0.0
+ *
+ * */
+class DataDictIndex extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      filter: {}
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrive data from the provided URL and save it in state
+   * Additionaly add hiddenHeaders to global loris vairable
+   * for easy access by columnFormatter.
+   */
+  fetchData() {
+    $.ajax(this.props.DataURL, {
+      method: "GET",
+      dataType: 'json',
+      success: function(data) {
+        loris.hiddenHeaders = data.hiddenHeaders ? data.hiddenHeaders : [];
+        this.setState({
+          Data: data,
+          isLoaded: true
+        });
+      }.bind(this),
+      error: function(error) {
+        console.error(error);
+      }
+    });
+  }
+
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  resetFilters() {
+    this.refs.dataDictFilter.clearFilter();
+  }
+
+  render() {
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return (
+        <button className="btn-info has-spinner">
+          Loading
+          <span
+          className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
+        </span>
+        </button>
+      );
+    }
+
+    return (
+        <div>
+            <FilterForm
+                Module="datadict"
+                name="data_dict_filter"
+                id="data_dict_filter"
+                ref="dataDictFilter"
```

Comment:
assigning strings to ref props is legacy now. i think this would be a good place to start using the [callback pattern](https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs).
except the arrow function pattern they use in that link will fail lint :put_litter_in_its_place:, so you will need to use the traditional function syntax